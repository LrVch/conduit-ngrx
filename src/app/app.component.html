<div [class]="'theme-wrapper ' + (effectiveTheme$ | async)"
     [ngClass]="{'sticky-header': (stickyHeader$ | async)}">
  <mat-sidenav-container>
    <mat-sidenav [opened]="sideNavOpen$ | async"
                 (closed)="onCloseSideNave()"
                 [mode]="asideOpenMode$ | async">
      <app-aside *ngIf="(settings$ | async)"
                 [theme]="theme$ | async"
                 [themes]="themes$ | async"
                 [stickyHeader]="stickyHeader$ | async"
                 [routeAnimationsEnabled]="routeAnimationsEnabled$ | async"
                 [asideOpenMode]="asideOpenMode$ | async"
                 [asideOpenModes]="asideOpenModes$ | async"
                 [autoNightMode]="autoNightMode$ | async"
                 [autoNightModeFrom]="autoNightModeFrom$ | async"
                 [autoNightModeTo]="autoNightModeTo$ | async"
                 [routeAnimateonsChangeType]="routeAnimateonsChangeType$ | async"
                 [routeAnimateonsChangeTypes]="routeAnimateonsChangeTypes$ | async"
                 (changeRouteAnimationsType)="onChangeRouteAnimationsType($event)"
                 (changeStickyHeader)="onChangeStickyHeader($event)"
                 (changeAsideOpenMode)="onChangeAsideOpenMode($event)"
                 (changeAutoNightMode)="onChangeAutoNightMode($event)"
                 (changeAutoNightModeFrom)="onChangeAutoNightModeFrom($event)"
                 (changeAutoNightModeTo)="onCangeAutoNightModeTo($event)"
                 (changeRouteAnimationsEnable)="onChangeRouteAnimationsEnable($event)"
                 (changeTheme)="onChangeTheme($event)"></app-aside>
    </mat-sidenav>
    <mat-sidenav-content>

      <div class="layout">
        <div class="main">
          <div class="main-content">
            <app-header [user]="user$ | async"
                        [loggedIn]="loggedIn$ | async"
                        [showLoader]="showMailLoader$ | async"
                        [languages]="languages$ | async"
                        [language]="language$ | async"
                        (changeLanguage)="onChangeLanguage($event)"
                        (toggleSideNave)="onToggleSideNave()"></app-header>

            <div [@routeAnimation]="prepRouteState(routerOutlet)">
              <router-outlet #routerOutlet="outlet"></router-outlet>
            </div>
          </div>

          <app-footer></app-footer>
        </div>
      </div>
    </mat-sidenav-content>

  </mat-sidenav-container>

</div>