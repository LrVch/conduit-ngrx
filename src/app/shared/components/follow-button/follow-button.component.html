<button mat-stroked-button
        [disabled]="isLoading"
        [color]="followed ? '': 'primary'"
        (click)="onFavoriteClick($event)">
  {{ followed ? ('conduit.followButton.unfollow' | translate) : ('conduit.followButton.follow' | translate) }}
  <ng-content></ng-content>
</button>