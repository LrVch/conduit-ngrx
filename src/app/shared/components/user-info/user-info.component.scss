.user-info {
  padding-top: 50px;
  @media (max-width: 768px) {
    padding-top: 70px;
  }
  @media (max-width: 590px) {
    padding-top: 60px;
  }
  @media (max-width: 575px) {
    padding-top: 30px;
  }
}

.user-info__avatar {
  display: flex;
  overflow: hidden;
  img {
    max-width: 100%;
  }
}
