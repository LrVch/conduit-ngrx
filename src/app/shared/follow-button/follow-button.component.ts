import { Component, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-follow-button',
  templateUrl: './follow-button.component.html',
  styleUrls: ['./follow-button.component.scss']
})
export class FollowButtonComponent {
  @Input('followed') followed = false;
  @Input('isLoading') isLoading = false;
  @Output() followedToggle = new EventEmitter();

  onFavoriteClick($event): void {
    $event.preventDefault();
    if (this.isLoading) {
      return;
    }

    this.followedToggle.emit();
  }
}
