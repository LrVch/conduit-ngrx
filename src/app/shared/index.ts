export * from './article/article.component';
export * from './articles-list/articles-list.component';
export * from './content-tabs/content-tabs.component';
export * from './tags/tags.component';
export * from './favorite/favorite.component';
export * from './layout';
export * from './shared.module';
export * from './utils';
export * from './list-error/list-error.component';
export * from './show-authed.directive';
export * from './user-info/user-info.component';
export * from './follow-button/follow-button.component';
export * from './tag-list/tag-list.component';
export * from './comments/comments.component';
export * from './base-from/base-from.component';
