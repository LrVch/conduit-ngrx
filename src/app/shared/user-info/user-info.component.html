<div class="user-info">
  <div class="container">
    <div class="row">

      <div class="col-xs-12 col-md-10 offset-md-1">
        <mat-card>
          <mat-card-header>
            <div mat-card-avatar
                 class="user-info__avatar">
              <img [src]="profile.image"
                   [attr.alt]="profile.username"></div>
            <mat-card-title>{{profile.username}}</mat-card-title>
            <mat-card-subtitle>{{profile.bio}} &nbsp;</mat-card-subtitle>
          </mat-card-header>
          <mat-card-content>
            <button routerLink="/settings"
                    [hidden]="!isUser"
                    mat-stroked-button> Edit Profile Settings</button>
            <app-follow-button [hidden]="isUser"
                               [isLoading]="profile.isFollowLoading"
                               [followed]="profile.following"
                               (followedToggle)="onFollowedToggle(profile)">
              {{ profile.username }}
            </app-follow-button>
          </mat-card-content>
        </mat-card>

      </div>

    </div>
  </div>
</div>