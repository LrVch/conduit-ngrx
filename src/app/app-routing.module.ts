import { NgModule } from '@angular/core';
import { Routes, RouterModule, PreloadAllModules } from '@angular/router';
import { PageNotFoundComponent } from './shared';
import { AuthGuard } from './core';
const routes: Routes = [
  {
    path: 'settings',
    loadChildren: './settings/settings.module#SettingsModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'profile',
    loadChildren: './profile/profile.module#ProfileModule',
  },
  {
    path: 'editor',
    loadChildren: './editor/editor.module#EditorModule',
    canLoad: [AuthGuard],
  },
  {
    path: 'article',
    loadChildren: './article/article.module#ArticleModule',
  },
  {
    path: '**',
    component: PageNotFoundComponent,
    data: { title: 'conduit.menu.notfound' }
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes, {
    scrollPositionRestoration: 'enabled'
    // preload all modules; optionally we could
    // implement a custom preloading strategy for just some
    // of the modules (PRs welcome ðŸ˜‰)
    // preloadingStrategy: PreloadAllModules
  })],
  exports: [RouterModule]
})
export class AppRoutingModule { }
