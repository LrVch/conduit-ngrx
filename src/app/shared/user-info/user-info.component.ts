import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Profile } from 'src/app/core';

@Component({
  selector: 'app-user-info',
  templateUrl: './user-info.component.html',
  styleUrls: ['./user-info.component.scss']
})
export class UserInfoComponent {
  @Input('profile') profile: Profile = {} as Profile;
  @Input('isUser') isUser = false;
  @Output() folowing = new EventEmitter<Profile>();


  onFollowedToggle(profile: Profile) {
    this.folowing.emit(profile);
  }
}
