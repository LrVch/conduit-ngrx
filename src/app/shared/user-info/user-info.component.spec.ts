import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '../../lib/testing';

import { UserInfoComponent } from './user-info.component';
import { NO_ERRORS_SCHEMA, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { Profile } from 'src/app/core';
import { FollowButtonComponent } from '../follow-button/follow-button.component';

describe('UserInfoComponent', () => {
    let component: UserInfoComponent;
    let fixture: ComponentFixture<UserInfoComponent>;
    let de: DebugElement;
    let el: HTMLElement;
    const initialProfile: Profile  = {} as Profile;

    const profile: Profile  = {
        username: 'username',
        bio: 'bio',
        image: 'image',
        following: false,
        isFollowLoading: false
    };

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [UserInfoComponent, FollowButtonComponent],
                    schemas: [NO_ERRORS_SCHEMA]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(UserInfoComponent);
                component = fixture.componentInstance;
                component.profile = initialProfile;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has "profile" @Input', () => {
        expect(component.profile).toEqual({});
    });

    it('should has "isUser" @Input', () => {
        expect(component.isUser).toEqual(false);
    });

    it('should compile', () => {
        component.profile = profile;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should show edit profile settings button by "isUser" @Input equals to true', () => {
        component.profile = profile;
        component.isUser = true;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should show following button by "isUser" @Input equals to false', () => {
        component.profile = profile;
        component.isUser = false;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should raise folowing event when clicked (onFollowedToggle) and pass the profile', () => {
        let toggledProfile = null;
        component.profile = profile;
        component.folowing.subscribe((p: Profile) => {
            toggledProfile = p;
        });

        fixture.detectChanges();

        const followButton = de.query(By.css('app-follow-button')).query(By.css('button'));
        const event = {
            preventDefault() {}
        };

        followButton.triggerEventHandler('click', event);

        expect(toggledProfile).toBe(profile);
    });
});

