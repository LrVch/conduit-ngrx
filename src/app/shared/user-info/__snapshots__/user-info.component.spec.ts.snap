// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoComponent should compile 1`] = `
<app-user-info
  folowing={[Function EventEmitter]}
  isUser="false"
  profile={[Function Object]}
>
  <div
    class="user-info"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-12 col-md-10 offset-md-1"
        >
          <mat-card>
            <mat-card-header>
              <div
                class="user-info__avatar"
                mat-card-avatar=""
              >
                <img
                  alt="username"
                  src="image"
                />
              </div>
              <mat-card-title>
                username
              </mat-card-title>
              <mat-card-subtitle>
                bio  
              </mat-card-subtitle>
            </mat-card-header>
            <mat-card-content>
              <button
                hidden=""
                mat-stroked-button=""
                routerlink="/settings"
              >
                 Edit Profile Settings
              </button>
              <app-follow-button
                ng-reflect-followed="false"
                ng-reflect-is-loading="false"
              >
                <button
                  mat-stroked-button=""
                >
                   Follow  
                   username 
                </button>
              </app-follow-button>
            </mat-card-content>
          </mat-card>
        </div>
      </div>
    </div>
  </div>
</app-user-info>
`;

exports[`UserInfoComponent should show edit profile settings button by "isUser" @Input equals to true 1`] = `
<app-user-info
  folowing={[Function EventEmitter]}
  isUser={[Function Boolean]}
  profile={[Function Object]}
>
  <div
    class="user-info"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-12 col-md-10 offset-md-1"
        >
          <mat-card>
            <mat-card-header>
              <div
                class="user-info__avatar"
                mat-card-avatar=""
              >
                <img
                  alt="username"
                  src="image"
                />
              </div>
              <mat-card-title>
                username
              </mat-card-title>
              <mat-card-subtitle>
                bio  
              </mat-card-subtitle>
            </mat-card-header>
            <mat-card-content>
              <button
                mat-stroked-button=""
                routerlink="/settings"
              >
                 Edit Profile Settings
              </button>
              <app-follow-button
                hidden=""
                ng-reflect-followed="false"
                ng-reflect-is-loading="false"
              >
                <button
                  mat-stroked-button=""
                >
                   Follow  
                   username 
                </button>
              </app-follow-button>
            </mat-card-content>
          </mat-card>
        </div>
      </div>
    </div>
  </div>
</app-user-info>
`;

exports[`UserInfoComponent should show following button by "isUser" @Input equals to false 1`] = `
<app-user-info
  folowing={[Function EventEmitter]}
  isUser="false"
  profile={[Function Object]}
>
  <div
    class="user-info"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-12 col-md-10 offset-md-1"
        >
          <mat-card>
            <mat-card-header>
              <div
                class="user-info__avatar"
                mat-card-avatar=""
              >
                <img
                  alt="username"
                  src="image"
                />
              </div>
              <mat-card-title>
                username
              </mat-card-title>
              <mat-card-subtitle>
                bio  
              </mat-card-subtitle>
            </mat-card-header>
            <mat-card-content>
              <button
                hidden=""
                mat-stroked-button=""
                routerlink="/settings"
              >
                 Edit Profile Settings
              </button>
              <app-follow-button
                ng-reflect-followed="false"
                ng-reflect-is-loading="false"
              >
                <button
                  mat-stroked-button=""
                >
                   Follow  
                   username 
                </button>
              </app-follow-button>
            </mat-card-content>
          </mat-card>
        </div>
      </div>
    </div>
  </div>
</app-user-info>
`;
